### PR TITLE
feat: add warning "cannot be a function" in getValueProps

### PR DESCRIPTION
**Pull request by LongHaoo**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [x] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

关联 PR https://github.com/ant-design/ant-design/pull/46445

关联 issue https://github.com/ant-design/ant-design/issues/46417

### 💡 Background and solution

根据 #46445  的改动为 getValueProps 的返回值加上一层 warning，判断返回值是否为 function 是 function 则给予提示

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Add a prompt for the error message returned by the Field component's getValueProps      |
| 🇨🇳 Chinese |     为  Field 组件的 getValueProps  返回错误提示增加提示  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: src/Field.tsx [hunk 1]
```diff
@@ -578,9 +578,16 @@ class Field extends React.Component<InternalFieldProps, FieldState> implements F
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     const originTriggerFunc: any = childProps[trigger];
 
+    const valueProps = mergedGetValueProps(value);
+
+    // warning when prop value is function
```
Comment (zombieJ) [inline src/Field.tsx:583]: 用 `process`  包一下，免得打包后还会检查。

Comment (LongHaoo) [reply src/Field.tsx:583]: 有无事例

File: src/Field.tsx [hunk 2]
```diff
@@ -578,9 +578,16 @@ class Field extends React.Component<InternalFieldProps, FieldState> implements F
...
+    Object.keys(valueProps).forEach(key => {
+      warning(typeof valueProps[key] !== 'function', `cannot return prop value is function in getValueProps (prop name: "${key}")`)
```
Comment (zombieJ) [inline src/Field.tsx:585]: 这个描述比较怪异，它不是不能返回。而是不推荐通过 `getValueProps` 动态添加 function。

Comment (LongHaoo) [reply src/Field.tsx:585]: 好的👌

File: src/Field.tsx [hunk 3]
```diff
@@ -578,9 +578,18 @@ class Field extends React.Component<InternalFieldProps, FieldState> implements F
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     const originTriggerFunc: any = childProps[trigger];
 
+    const valueProps = mergedGetValueProps(value);
+
+    // warning when prop value is function
+    if (process.env.NODE_ENV !== 'production') {
+      Object.keys(valueProps).forEach(key => {
+        warning(typeof valueProps[key] !== 'function', `Not recommended return prop value is function in getValueProps (prop name: "${key}")`)
```
Comment (zombieJ) [inline src/Field.tsx:586]: > It's not recommended to generate dynamic function prop by `getValueProps`. Please pass it to child component directly.

File: src/Field.tsx [hunk 4]
```diff
@@ -578,9 +578,18 @@ class Field extends React.Component<InternalFieldProps, FieldState> implements F
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     const originTriggerFunc: any = childProps[trigger];
 
+    const valueProps = mergedGetValueProps(value);
+
+    // warning when prop value is function
+    if (process.env.NODE_ENV !== 'production') {
+      Object.keys(valueProps).forEach(key => {
+        warning(typeof valueProps[key] !== 'function', `It's not recommended to generate dynamic function prop by \`getValueProps\`. Please pass it to child component directly (prop: ${key})`)
```
Comment (crazyair) [inline src/Field.tsx:586]: 一直不知道 warning 第一个参数是 true 打印，还是 false 打印

Comment (LongHaoo) [reply src/Field.tsx:586]: false hh 我也是看下面代码学的，下面代码也有用 warning 的

Comment (crazyair) [reply src/Field.tsx:586]: 所以我倾向
```tsx
if(xx){
   warning(false,'xx')
}
```

**Pull request merged**